src/vivarium_census_prl_synth_pop/components/person.py

```diff
@@ -130,59 +132,55 @@ def on_time_step(self, event: Event) -> None:
         new_household_data_map = new_household_data.set_index("household_id")
 
         # update household data for domestic movers
-        domestic_movers["household_id"] = new_households
-        domestic_movers = update_address_and_zipcode(
-            df=domestic_movers,
-            rows_to_update=domestic_movers.index,
-            id_key=domestic_movers["household_id"],
+        persons.loc[domestic_movers_idx, "household_id"] = new_households
+        persons = update_address_and_zipcode(
+            df=persons,
+            rows_to_update=domestic_movers_idx,
+            id_key=new_households,
             address_map=new_household_data_map["address"],
             zipcode_map=new_household_data_map["zipcode"],
         )
 
         # update relation to head of household data
-        domestic_movers["relation_to_household_head"] = "Other nonrelative"
-        domestic_movers.loc[
-            domestic_movers["household_id"].isin(
-                data_values.NONINSTITUTIONAL_GROUP_QUARTER_IDS.values()
-            ),
+        persons.loc[domestic_movers_idx, "relation_to_household_head"] = "Other nonrelative"
+        persons.loc[
+            (persons.index.isin(domestic_movers_idx)) &
+            (persons["household_id"].isin(
+                data_values.NONINSTITUTIONAL_GROUP_QUARTER_IDS.values())),
             "relation_to_household_head",
         ] = "Noninstitutionalized GQ pop"
-        domestic_movers.loc[
-            domestic_movers["household_id"].isin(
-                data_values.INSTITUTIONAL_GROUP_QUARTER_IDS.values()
-            ),
+        persons.loc[
+            (persons.index.isin(domestic_movers_idx)) &
+            (persons["household_id"].isin(
+                data_values.INSTITUTIONAL_GROUP_QUARTER_IDS.values())),
             "relation_to_household_head",
         ] = "Institutionalized GQ pop"
 
         # Update housing type
-        domestic_movers.loc[
-            domestic_movers["household_id"].isin(data_values.HOUSING_TYPE_MAP.keys()),
+        persons.loc[
+            (persons.index.isin(domestic_movers_idx)) &
+            (persons["household_id"].isin(data_values.HOUSING_TYPE_MAP.keys())),
```

Comment:
Can you update the variable name `HOUSING_TYPE_MAP` to contain some reference to GQ, since these are only GQ housing types?

Reply:
Will do.